src/meta/meta_ckpt.c

```diff
@@ -1262,6 +1262,195 @@ __wt_meta_ckptlist_update_config(
     return (ret);
 }
 
+/*
+ * __ckpt_extract_blkmod_bitmap --
+ *     Extracts the blkmod info in a checkpoint into a bitmap in a WT_ITEM.
+ */
+static int
+__ckpt_extract_blkmod_bitmap(
+  WT_SESSION_IMPL *session, WT_CKPT *ckpt, const char *id, WT_ITEM *output_bitmap)
+{
+    WT_BLOCK_MODS *blk;
+    u_int i;
+
+    WT_CLEAR(*output_bitmap);
+
+    for (i = 0, blk = &ckpt->backup_blocks[0]; i < WT_BLKINCR_MAX; ++i, ++blk)
+        if (F_ISSET(blk, WT_BLOCK_MODS_VALID) && strcmp(id, blk->id_str) == 0)
+            WT_RET(__wt_buf_set(session, output_bitmap, blk->bitstring.data, blk->bitstring.size));
+
+    return (0);
+}
+
+/*
+ * __get_blkmods --
+ *     Extracts the blkmod info from the metadata about a file+id into a bitmap in a WT_ITEM.
+ */
+static WT_INLINE int
+__get_blkmods(WT_SESSION_IMPL *session, const char *uri, const char *id, WT_ITEM *output_item)
+{
+    WT_CONFIG blkconf;
+    WT_CONFIG_ITEM backup_config_value, blocks, blocks_key, blocks_value, key;
+    WT_CURSOR *metadata_cursor;
+    WT_DECL_RET;
+    char *file_config;
+
+    WT_CLEAR(backup_config_value);
+    WT_CLEAR(blocks);
+    WT_CLEAR(blocks_key);
+    WT_CLEAR(blocks_value);
+    WT_CLEAR(key);
+    WT_CLEAR(*output_item);
+
+    metadata_cursor = NULL;
+    file_config = NULL;
+
+    WT_RET(__wt_metadata_cursor_open(session, NULL, &metadata_cursor));
+
+    metadata_cursor->set_key(metadata_cursor, uri);
+    WT_ERR(metadata_cursor->search(metadata_cursor));
+    WT_ERR(metadata_cursor->get_value(metadata_cursor, &file_config));
+    WT_ERR(
+      __wt_config_getones(session, file_config, "checkpoint_backup_info", &backup_config_value));
+
+    if (backup_config_value.len > 0) {
```

Comment:
Should we early exit here by jumping to the err label instead of having a double indent?
```
if (backup_config_value.len == 0 || backup_config_value.type != WT_CONFIG_ITEM_STRUCT)
     goto done;
```
Note that `done` is not existing, we could do `goto err;` but it's not an error... So it could be misleading.

Reply:
Hi @etienneptl , I prefer to avoid `goto` statements whenever possible, so I prefer the `if` statement as it is rather than adding another `goto`.